dialyzer/src/main.ts

```diff
@@ -92,15 +92,26 @@ async function run(): Promise<void> {
   } = await elixirVersions();
   const mixLockHash = await hashFiles(["mix.lock", "apps/*/mix.lock"]);
   const dialyzerPaths = [" _build/*/*.plt*"];
-  const cacheKey = `${architecture}-dialyzer-${otp_release}-${erts_version}-${elixir_version}-${mixLockHash}`;
-  const restoreKeys = [
-    `${architecture}-dialyzer-${otp_release}-${erts_version}-${elixir_version}-`,
-    `${architecture}-dialyzer-${otp_release}-${erts_version}-`,
-    // previous version of the Dialyzer cache
-    `${architecture}-dialyzer-${otp_release}-${elixir_version}-`,
-    `${architecture}-dialyzer-${otp_release}-`,
-    `${architecture}-dialyzer-`,
-  ];
+  const cacheKeyVersion = core.getInput("cache-key-version");
+  let cacheKey = `${architecture}-dialyzer-${otp_release}-${erts_version}-${elixir_version}-${mixLockHash}`;
+  if (cacheKeyVersion !== "") {
+    cacheKey = `${cacheKeyVersion}-${cacheKey}`;
+  }
+
+  const shouldUseFallbackCacheKeys =
+    core.getInput("use-fallback-cache-keys") === "true";
+  console.log("Using fallback cache keys?", shouldUseFallbackCacheKeys);
+  const restoreKeys = shouldUseFallbackCacheKeys
+    ? [
+        `${architecture}-dialyzer-${otp_release}-${erts_version}-${elixir_version}-`,
```

Comment:
Question: did you mean to include the cacheKey in these restoreKeys?

Reply:
Good catch! I also thought about removing the restore keys if they could lead to caching issues like Draft was seeing. Should restoreKeys still be used by default if they can lead to bad cached states?

Reply:
I haven't heard of any other projects running into this issue, and normally the Dialyzer cache works even after upgrading packages (the main reason the PLT is updated) so I think we should keep them for now.

If we do continue to see issues with other projects, we can always come back and remove these fallbacks.